docs/conf.py

```diff
@@ -272,10 +272,31 @@
     "https://github.com/pypa/twine/issues/*",
     # Avoid errors from channels interpreted as anchors
     "https://web.libera.chat/#",
+    # Avoid error from InvalidPyPIUploadURL docstring
+    "https://upload.pypi.org/legacy",
 ]
 
 # Example configuration for intersphinx: refer to the Python standard library.
 intersphinx_mapping = {"https://docs.python.org/": None}
 
 # Be strict about the invalid references:
 nitpicky = True
+
+# TODO: Try to add these to intersphinx_mapping
+nitpick_ignore_regex = [
+    (r"py:.*", r"(pkginfo|requests|IO).*"),
```

Comment:
Why can't we add the intersphinx mapping for requests? It should be as simple as updating 280 to be 

```py
{
  "python": ("https://docs.python.org/", None),
  "requests": ("https://requests.readthedocs.io/", None),
}
```

See also: https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html

Reply:
I tried that, via:

```python
intersphinx_mapping = {
    "python": ("https://docs.python.org/", None),
    "requests": ("https://docs.python-requests.org/en/latest/", None),
}
```

But it's not entirely working:

```
WARNING: py:class reference target not found: requests.models.Response
```

Via [intersphinx-untangled](https://webknjaz.github.io/intersphinx-untangled/docs.python-requests.org/), it seems that the documented reference is the public `requests.Response`, and even though that's the type hint, it's being resolved to the internal class definition.

Reply:
Why can't we just use `requests.models.Response`? It's documented as that in the docs. Most people use `requests.Response` because the author wanted to re-export the world which was a bad idea anyway

Reply:
> Why can't we just use `requests.models.Response`? It's documented as that in the docs.

Where do you see it documented that way? I only see https://docs.python-requests.org/en/master/api/#requests.Response.

Reply:
I went down the rabbit hole on this, with the end result being https://github.com/sphinx-doc/sphinx/issues/9653.